app/assets/scripts/components/map.js

```diff
@@ -1,30 +1,83 @@
 'use strict';
 import React from 'react';
-import { connect } from 'react-redux';
 import window from 'global/window';
+import bbox from '@turf/bbox';
+const L = window.L;
 
 const tileLayer = 'https://api.mapbox.com/styles/v1/map-egypt/civld9uy0000n2kmnd7lqs3ne/tiles/256/{z}/{x}/{y}?access_token=pk.eyJ1IjoibWFwLWVneXB0IiwiYSI6ImNpdmxkMjl6bTA3c2YyeXBvNDJnZDlqZGMifQ.KQSizb18ILr6wri0cBcd2Q';
 
+const BOUNDS = [
+  [22.278144, 25.127830],
+  [31.118067, 33.719138]
+];
+
+function getLatLngBounds (bounds) {
+  let b = bbox(bounds);
+  let result = [[b[1], b[0]], [b[3], b[2]]];
+  return result;
+}
+
 const Map = React.createClass({
   displayName: 'Map',
 
   propTypes: {
+    location: React.PropTypes.object,
+    markers: React.PropTypes.array
+  },
+
+  fitMap: function (props) {
+    let bounds;
+    if (props.location) {
+      bounds = props.location.type
+        ? getLatLngBounds(props.location)
+        : props.location;
+    } else {
+      bounds = BOUNDS;
+    }
+    this.map.fitBounds(bounds);
+  },
+
+  addClusterMarkers: function (markers) {
+    const markerLayer = this.markerLayer;
+    markerLayer.clearLayers();
+    markers.forEach(function (marker) {
+      const leafletMarker = L.marker(marker.centroid, {
+        icon: L.mapbox.marker.icon({'marker-symbol': 'garden', 'marker-color': '0044FF'})
+      });
+      leafletMarker.bindPopup(
+        `<div class='marker__internal'>` +
```

Comment:
@derek for future reference, template strings can be multiline. No need to use a plus sign https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals